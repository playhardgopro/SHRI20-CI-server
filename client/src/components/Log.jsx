import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import './scss/Log.scss'

const Convert = require('ansi-to-html')

const convert = new Convert({
  fg: '#FFF',
  bg: '#000',
  newline: false,
  escapeXML: false,
  stream: false,
})

const cn = withNaming({ n: '', e: '__', m: '_' })

const Log = ({ children, className }) => {
  const cnLog = cn('log')(className)
  const log = `"> shri-2020-task-1@1.0.0 build \n> webpack --mode=production \"--colors\"\n\nHash: \u001b[1me541e85eb5e88a853b17\u001b[39m\u001b[22m\nVersion: webpack \u001b[1m4.41.5\u001b[39m\u001b[22m\nTime: \u001b[1m1782\u001b[39m\u001b[22mms\nBuilt at: 2020-03-30 \u001b[1m0:39:32\u001b[39m\u001b[22m\n        \u001b[1mAsset\u001b[39m\u001b[22m      \u001b[1mSize\u001b[39m\u001b[22m  \u001b[1mChunks\u001b[39m\u001b[22m  \u001b[1m\u001b[39m\u001b[22m                 \u001b[1m\u001b[39m\u001b[22m\u001b[1mChunk Names\u001b[39m\u001b[22m\n    \u001b[1m\u001b[32mscript.js\u001b[39m\u001b[22m  1.83 KiB       \u001b[1m0\u001b[39m\u001b[22m  \u001b[1m\u001b[32m[emitted]\u001b[39m\u001b[22m        main\n\u001b[1m\u001b[32mscript.js.map\u001b[39m\u001b[22m  7.03 KiB       \u001b[1m0\u001b[39m\u001b[22m  \u001b[1m\u001b[32m[emitted] [dev]\u001b[39m\u001b[22m  main\n    \u001b[1m\u001b[32mstyle.css\u001b[39m\u001b[22m  20.5 KiB       \u001b[1m0\u001b[39m\u001b[22m  \u001b[1m\u001b[32m[emitted]\u001b[39m\u001b[22m        main\n\u001b[1m\u001b[32mstyle.css.map\u001b[39m\u001b[22m  24.8 KiB       \u001b[1m0\u001b[39m\u001b[22m  \u001b[1m\u001b[32m[emitted] [dev]\u001b[39m\u001b[22m  main\nEntrypoint \u001b[1mmain\u001b[39m\u001b[22m = \u001b[1m\u001b[32mstyle.css\u001b[39m\u001b[22m \u001b[1m\u001b[32mscript.js\u001b[39m\u001b[22m \u001b[1m\u001b[32mstyle.css.map\u001b[39m\u001b[22m \u001b[1m\u001b[32mscript.js.map\u001b[39m\u001b[22m\n[0] \u001b[1m./src/js/utils.js\u001b[39m\u001b[22m 514 bytes {\u001b[1m\u001b[33m0\u001b[39m\u001b[22m}\u001b[1m\u001b[32m [built]\u001b[39m\u001b[22m\n[1] \u001b[1mmulti ./src/js/_index.js ./src/scss/index.scss\u001b[39m\u001b[22m 40 bytes {\u001b[1m\u001b[33m0\u001b[39m\u001b[22m}\u001b[1m\u001b[32m [built]\u001b[39m\u001b[22m\n[2] \u001b[1m./src/js/_index.js\u001b[39m\u001b[22m 89 bytes {\u001b[1m\u001b[33m0\u001b[39m\u001b[22m}\u001b[1m\u001b[32m [built]\u001b[39m\u001b[22m\n[3] \u001b[1m./src/js/onoffswitch.js\u001b[39m\u001b[22m 654 bytes {\u001b[1m\u001b[33m0\u001b[39m\u001b[22m}\u001b[1m\u001b[32m [built]\u001b[39m\u001b[22m\n[4] \u001b[1m./src/js/e-accordion.js\u001b[39m\u001b[22m 403 bytes {\u001b[1m\u001b[33m0\u001b[39m\u001b[22m}\u001b[1m\u001b[32m [built]\u001b[39m\u001b[22m\n[5] \u001b[1m./src/scss/index.scss\u001b[39m\u001b[22m 39 bytes {\u001b[1m\u001b[33m0\u001b[39m\u001b[22m}\u001b[1m\u001b[32m [built]\u001b[39m\u001b[22m\n    + 1 hidden module\nChild \u001b[1mmini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!node_modules/postcss-loader/src/index.js!node_modules/sass-loader/dist/cjs.js!src/scss/index.scss\u001b[39m\u001b[22m:\n    Entrypoint \u001b[1mmini-css-extract-plugin\u001b[39m\u001b[22m = \u001b[1m\u001b[32m*\u001b[39m\u001b[22m\n    [0] \u001b[1m./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src!./node_modules/sass-loader/dist/cjs.js!./src/scss/index.scss\u001b[39m\u001b[22m 20.7 KiB {\u001b[1m\u001b[33m0\u001b[39m\u001b[22m}\u001b[1m\u001b[32m [built]\u001b[39m\u001b[22m\n        + 1 hidden module\n"`

  console.log(convert.toHtml(log))

  return (
    <div className="log">
      <div className="log__pre log__pre_scroll">
        <pre className="pre">{convert.toHtml(log)}</pre>
      </div>
    </div>
  )
}

Log.propTypes = {
  children: PropTypes.node,
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
}
Log.defaultProps = {
  children: '',
  className: {},
}

export default Log
