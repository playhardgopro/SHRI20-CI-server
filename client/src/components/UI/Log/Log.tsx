import React, { useEffect } from 'react'
import { withNaming } from '@bem-react/classname'
import './Log.scss'
import { useSelector } from 'react-redux'

import Convert from 'ansi-to-html'

const convert = new Convert({
  fg: '#000',
  bg: '#000',
  newline: false,
  escapeXML: false,
  stream: false,
})

const cn = withNaming({ n: '', e: '__', m: '_' })

interface LogProps {
  className?: {}
}

const Log: React.FC<LogProps> = () => {
  const cnLog = cn('log')
  const logText = useSelector((state: RootState) => state.build.logs) || 'Please, wait, until build process ends'

  return (
    <div className={cnLog()}>
      <div className={cnLog('pre', { scroll: true })}>
        <pre className="pre">
          <div dangerouslySetInnerHTML={{ __html: convert.toHtml(logText) }} />
        </pre>
      </div>
    </div>
  )
}

export default Log
