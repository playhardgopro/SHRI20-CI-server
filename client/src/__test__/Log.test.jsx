import React from 'react'
import renderer from 'react-test-renderer'
import Log from '../components/UI/Log/Log'

describe('Log', () => {
  it('renders correctly', () => {
    const log = renderer.create(<Log />).toJSON()

    expect(log).toMatchSnapshot()
  })
})
