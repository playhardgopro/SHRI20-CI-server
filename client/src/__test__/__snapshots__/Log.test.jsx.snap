// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log renders correctly 1`] = `
<div
  className="log"
>
  <div
    className="log__pre log__pre_scroll"
  >
    <pre
      className="pre"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "> shri-2020-task-1@1.0.0 build 
> webpack --mode=production \\"--colors\\"

Hash: <b>e541e85eb5e88a853b17<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
Version: webpack <b>4.41.5<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
Time: <b>1782<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">ms
Built at: 2020-03-30 <b>0:39:32<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
        <b>Asset<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">      <b>Size<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  <b>Chunks<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  <b><span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">                 <b><span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"><b>Chunk Names<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
    <b><span style=\\"color:#0A0\\">script.js<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  1.83 KiB       <b>0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  <b><span style=\\"color:#0A0\\">[emitted]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">        main
<b><span style=\\"color:#0A0\\">script.js.map<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  7.03 KiB       <b>0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  <b><span style=\\"color:#0A0\\">[emitted] [dev]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  main
    <b><span style=\\"color:#0A0\\">style.css<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  20.5 KiB       <b>0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  <b><span style=\\"color:#0A0\\">[emitted]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">        main
<b><span style=\\"color:#0A0\\">style.css.map<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  24.8 KiB       <b>0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  <b><span style=\\"color:#0A0\\">[emitted] [dev]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">  main
Entrypoint <b>main<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> = <b><span style=\\"color:#0A0\\">style.css<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> <b><span style=\\"color:#0A0\\">script.js<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> <b><span style=\\"color:#0A0\\">style.css.map<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> <b><span style=\\"color:#0A0\\">script.js.map<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
[0] <b>./src/js/utils.js<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> 514 bytes {<b><span style=\\"color:#A50\\">0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">}<b><span style=\\"color:#0A0\\"> [built]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
[1] <b>multi ./src/js/_index.js ./src/scss/index.scss<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> 40 bytes {<b><span style=\\"color:#A50\\">0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">}<b><span style=\\"color:#0A0\\"> [built]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
[2] <b>./src/js/_index.js<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> 89 bytes {<b><span style=\\"color:#A50\\">0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">}<b><span style=\\"color:#0A0\\"> [built]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
[3] <b>./src/js/onoffswitch.js<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> 654 bytes {<b><span style=\\"color:#A50\\">0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">}<b><span style=\\"color:#0A0\\"> [built]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
[4] <b>./src/js/e-accordion.js<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> 403 bytes {<b><span style=\\"color:#A50\\">0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">}<b><span style=\\"color:#0A0\\"> [built]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
[5] <b>./src/scss/index.scss<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> 39 bytes {<b><span style=\\"color:#A50\\">0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">}<b><span style=\\"color:#0A0\\"> [built]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
    + 1 hidden module
Child <b>mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!node_modules/postcss-loader/src/index.js!node_modules/sass-loader/dist/cjs.js!src/scss/index.scss<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">:
    Entrypoint <b>mini-css-extract-plugin<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> = <b><span style=\\"color:#0A0\\">*<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
    [0] <b>./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src!./node_modules/sass-loader/dist/cjs.js!./src/scss/index.scss<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\"> 20.7 KiB {<b><span style=\\"color:#A50\\">0<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">}<b><span style=\\"color:#0A0\\"> [built]<span style=\\"color:#000\\"><span style=\\"font-weight:normal;text-decoration:none;font-style:normal\\">
        + 1 hidden module
</span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></span></b></span></span></b></span></span></span></b></span></span></b></span></span></b></span></span></b></span></span></b></span></span></b></span></span></b></span></span></b></span></span></b></span></span></b></span></span></b>",
          }
        }
      />
    </pre>
  </div>
</div>
`;
